include/Contact.php

```diff
@@ -662,4 +664,34 @@ function posts_from_contact($a, $contact_id) {
 
 	return $o;
 }
+
+/**
+ * @brief Returns a formatted location string from the given profile array
+ *
+ * @param array $profile Profile array (Generated from the "profile" table)
+ *
+ * @return string Location string
+ */
+function formatted_location($profile) {
```

Comment:
I have to write a test in on Wednesday. It is exams time at the end of the semester :-( Because I have no real good organization I don't really know how much time I will have in the next weeks. Maybe I have some time, maybe not :-/ 
But I have to say that I'm really interested in such things.

BTW: I wan't to give you respect for the large work on code cleaning. It's really awesome.

Ahh and if I find some time I will try to finish https://github.com/rabuzarus/friendica/tree/1901_autocomplete (I don't know if it is wanted. Originally it should be only a test. To sum up: It's a port of hubzillas autocomplete feature with the possibility to have autocomplete in the nav search for all gcontact contact entries)